### DISCUSSION TITLE
Shallow(er) Calibration Cycle

### DISCUSSION THREAD

**Discussion by cobras-ruck** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1206] [Ideas]
Hello and thanks for all the hard work on AlDente!

Out of curiosity, have shallower calibration cycles been tested before?

As an example, iOS has a native feature for limiting capacity to 80%. When in this mode, it periodically self-calibrates the battery, but does so by discharging to only ~70-75% - contrast this with AlDente, which discharges all the way down to ~10-15% when using the Calibration Mode feature.

A shallower calibration could further reduce battery wear and reduce both the impact of and time required for a calibration, offering more flexibility in scheduling and performing them. That said, I don't know if it would actually work without a deep discharge.

Thanks!

**Comment by MatthiasKerbl** [comment]
Hi @cobras-ruck,

Valid point. However, for the battery to get recalibrated, a full charge and discharge cycle is optimal.

Best,
Matthias

**Reply by cobras-ruck** [reply]
I'll keep experimenting with this.

I did a top-up and discharge, just 80-100-80, and the battery seems to have calibrated properly. It reported 6300 mAh prior to setting up AlDente with an 80% limit. After around a week it had fallen to 6250 mAh. After the "shallow" calibration it now reports 6300 mAh again.

Maybe we could consider an "experimental" or "advanced" feature to support customizing the calibration cycle like this?

Thanks again!

**Reply by MatthiasKerbl** [reply]
Hi @cobras-ruck,

Thanks for the update.
You can already configure a customized calibration cycle through the Apple Shortcuts integration or the schedule feature in AlDente.

Best,
Matthias

**Reply by cobras-ruck** [reply]
Thanks! Is this functionality new to 1.26? I'm running 1.25.2 Pro and while I can indeed schedule a calibration cycle, I don't see an option to actually configure the depth-of-discharge.

I had explored instead scheduling a Top-up overnight followed by a Discharge a few hours later. However, if I set "Start tasks at next opportunity" to ensure these happen, both run simultaneously upon waking up and the Discharge cancels the Top-up.

I also looked into using the Shortcuts to accomplish the same, but they don't seem to be automate-able like on iOS without using AppleScript or cron.

**Reply by MatthiasKerbl** [reply]
Hi @cobras-ruck,

You would need to configure it in Shortcuts or Schedule yourself.

If you are using the Schedule feature, you should activate Disable Sleep until Charge Limit as well and start the sequence when the MacBook is not in Sleep.

Best,
Matthias
